Add database

NastasyaRosh opened the pull request.

NastasyaRosh committed:
-подключила базу данных
-пока даоимпл есть только для фильмов, но не работает добавление фильма

NastasyaRosh committed:
-настроила дао для всего, кроме film, filmGenre и like

NastasyaRosh committed:
-сталось разобраться с genre и like

NastasyaRosh committed:
-сталось разобраться с like

NastasyaRosh committed:
-доделала логику, осталось вынести маперы и написать тесты

NastasyaRosh committed:
-готово, осталось пройтись удалить лишнее

NastasyaRosh committed:
-добавлена работа с базой данных

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -2,27 +2,33 @@
-    private final int duration;
+    private int duration;
     private Set<Integer> likes = new HashSet<>();
+    //private int rate;
```
В проекте не должно быть закомментированных строчек кода

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/dao/impl/LikeDaoImpl.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    private User userMapper(ResultSet resultSet) throws SQLException {
+        User userOut = new User();
```
В таких случаях можно создавать объект через билдер с помощью аннотации Lombok @Builder. Но это изменение по желанию, не настаиваю на нем.

NastasyaRosh replied:
Сделала во всех мапперах билдеры, так и впрямь красивее)

sergeymamonov replied:
можно еще посмотреть Mapstruct, так мапперы будут еще проще и красивее)

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -16,53 +21,86 @@ public class FilmService {
 
     private static final LocalDate FIRST_FILM = LocalDate.of(1895, 12, 28);
     private static final int DEFAULT_COUNT = 10;
```
Нужна ли в этом классе константа.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -16,53 +21,86 @@ public class FilmService {
 
     @Autowired
-    public FilmService(FilmStorage filmStorage) {
+    public FilmService(FilmDao filmStorage, FilmGenreDao filmGenreDao, LikeDaoImpl likeDao, UserService userService) {
```
Можно использовать аннотации Lombok, чтобы не писать вручную конструктор.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -16,53 +21,86 @@ public class FilmService {
+        return likeDao.findAllPopular(count);
     }
 
     private void checkId(int id) {
```
Используется ли где-то этот метод?

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -16,53 +21,86 @@ public class FilmService {
 
     private static final LocalDate FIRST_FILM = LocalDate.of(1895, 12, 28);
```
Отлично, что используете константы. Я бы только предложил немного изменить название. То есть нужно отображать в названии бизнес смысл. Поэтому такие константы часто именуют MAX_..., MIN_..., EARLIEST_... В данном случае эта дата важна с тем смыслом, что это дата выхода первого фильма, а самая ранняя допустимая дата. Но это изменение по желанию.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/GenreService.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    public Genre genreById(Integer id) throws FilmOrUserNotExist {
+        Genre genreOut = genreDao.genreById(id).orElseThrow(() ->
```
Можно сразу возвращать результат выражения без создания дополнительной переменной.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/RatingService.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    public Rating ratingById(Integer id) throws FilmOrUserNotExist {
+        Rating ratingOut = ratingDao.ratingById(id).orElseThrow(() ->
```
Аналогично. Можно сразу возвращать результат выражения без создания дополнительной переменной.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -29,44 +32,48 @@ public User addUser(User user) {
 
     public User userById(int id) throws FilmOrUserNotExist {
-        return userStorage.userById(id);
+        User userOut = userStorage.userById(id).orElseThrow(() ->
```
Можно сразу возвращать результат выражения без создания дополнительной переменной.

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -29,44 +32,48 @@ public User addUser(User user) {
+        return friendDao.commonFriends(id, otherId);
     }
 
     private void forNullName(User user) {
```
Методы должны начинаться с глагола. В данном случае можно назвать setFullName.

NastasyaRosh replied:
Исправила и у других методов названия тоже

NastasyaRosh committed:
-style убрала лишние переменные, методы и константы
-style изменила названия методов - теперь все их названия начинаются с глаголов
-refactor сделала создание конструкторов через аннотацию @RequiredArgsConstructor
-refactor сделала все мапперы через билдеры

sergeymamonov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -1,78 +1,102 @@
-    private static final LocalDate FIRST_FILM = LocalDate.of(1895, 12, 28);
-    private static final int DEFAULT_COUNT = 10;
-    private final FilmStorage filmStorage;
+    private static final LocalDate EARLIEST_FIRST_DATE = LocalDate.of(1895, 12, 28);
```
EARLIEST_RELEASE_DATE - так будет понятнее о какой дате идет речь.

NastasyaRosh committed:
-style изменила название константы

NastasyaRosh committed:
-style изменила название константы

NastasyaRosh committed:
-style изменила название константы

NastasyaRosh committed:
Merge remote-tracking branch 'origin/add-database' into add-database

NastasyaRosh committed:
Вернула и изменила имя константы

NastasyaRosh committed:
Merge remote-tracking branch 'origin/add-database' into add-database
